src/index.js

```diff
@@ -10,87 +10,102 @@ import './css/styles.css'
+  window.onpopsate = () => location.reload()
 
-function indicateLoadingFailure() {
-  // makes the loading state "indeterminate", like it's loading forever.
-  let progress = document.getElementById('loading-indicator')
-  progress.removeAttribute('value')
-  progress.classList.add(ERROR_CLASS)
-}
+  let csrfToken = document.querySelector('[name=csrfmiddlewaretoken]').value
+  orthography.registerEventListener(csrfToken)
 
-function hideLoadingIndicator() {
-  let progress = document.getElementById('loading-indicator')
-  progress.classList.remove(LOADING_CLASS, ERROR_CLASS)
-}
+  setupSearchBar()
+
+  let route = makeRouteRelativeToSlash(window.location.pathname)
+  // Tiny router.
+  if (route === '/') {
+    // Homepage
+    setSubtitle(null)
+  } else if (route === '/about') {
+    // About page
+    setSubtitle('About')
+  } else if (route == '/contact-us') {
+    // Contact page
+    setSubtitle('Contact us')
+  } else if (route === '/search') {
+    // Search page
+    prepareTooltips()
+  } else if (route.match(/^[/]word[/].+/)) {
+    // Word detail/paradigm page. This one has the 🔊 button.
+    setSubtitle(getEntryHead())
+    setupAudioOnPageLoad()
+  } else {
+    throw new Error(`Could not match route: ${route}`)
+  }
+})
 
 
+///////////////////////////// Internal functions /////////////////////////////
+
+function setupSearchBar() {
+  let searchBar = document.getElementById('search')
+  searchBar.addEventListener('input', () => {
+    let query = searchBar.value
+    loadSearchResults(searchBar)
+    changeTitleBySearchQuery(query)
+  })
+}
+
 /**
  * clean paradigm details
  */
 function cleanParadigm() {
-  $('#paradigm').remove()
+  removeElement(document.getElementById('paradigm'))
 }
 
-
 function showProse() {
-  let prose = $('#prose')
-  prose.show()
+  showElement(document.getElementById('prose'))
 }
 
 function hideProse() {
-  let prose = $('#prose')
-  prose.hide()
+  hideElement(document.getElementById('prose'))
 }
 
 /**
  * find #search-result-list element on the page to attach relevant handlers to the tooltip icons
  */
 function prepareTooltips() {
-  const $searchResultList = $('#search-result-list')
+  const searchResultList = getSearchResultList()
 
   // attach handlers for tooltip icon at preverb breakdown
-  $searchResultList.find('.definition-title__tooltip-icon').each(function () {
-    createTooltip($(this), $(this).next('.tooltip'))
-  })
-
-  // attach handlers for tooltip icon at preverb breakdown
-  $searchResultList.find('.preverb-breakdown__tooltip-icon').each(function () {
-    createTooltip($(this), $(this).next('.tooltip'))
-  })
+  let tooltips = searchResultList.querySelectorAll('.definition-title__tooltip-icon, .preverb-breakdown__tooltip-icon')
+  for (let icon of tooltips) {
+    createTooltip($(icon), $(icon).next('.tooltip'))
+  }
 }
 
 /**
- * use xhttp to load search results in place
+ * use AJAX to load search results in place
```

Comment:
@eddieantonio why is ajax favoured over http xhttp?

Reply:
AJAX refers to any _asynchronous_ HTTP request, whether it's done with the crusty old `XMLHttpRequest` API or the newer, less error-prone `fetch()` API.

Reply:
aah was asking why use fetch over XMLHttpRequest

Reply:
> aah was asking why use fetch over XMLHttpRequest

I've personally always went with XMLHttpRequest because it's what I was taught/learned first! Fetch is new to me but I'm aware that it is the future so I'll try to work at using it more go forward.

If I remember right, fetch handles some things better than XMLHttpRequest does (did?). I think I have a short tutorial tucked away for why the newer is better: just holler and I can pull it over the weekend or something 💪